designs/tripleo/osc-tripleo.md

```diff
@@ -0,0 +1,137 @@
+# OSC TripleO
+In order for OSC to be installable and configurable by [TripleO](#tripleo-home) a TripleO-Heat-Template (THT) must be created. This document outlines the details of this template highlighting OSC specific configuration and environment variables, TripleO required configuration, data persistence, TLS and high availability (HA) settings.   
+> Note: Details for HA, TLS and other advanced configuration will be added later. The first draft of this document covers required files, template parameters, environment variables and volumes.  
+
+## Assignees
+Emanoel Xavier - https://github.com/emanoelxavier
+
+## Background
+
+[TripleO](#tripleo-home) is the friendly name for “OpenStack on OpenStack”. It is an official OpenStack project with the goal of allowing you to deploy and manage a production cloud onto bare metal hardware using a subset of existing OpenStack components.  The deployed production cloud is called **overcloud** and the infrastructure services used for deployment are called **undercloud** as shown in the picture below. 
+
+![](./images/tripleo-services.png)  
+*TripleO Undercloud & Overcloud*  
+
+The intent of this work is to allow OSC to be more easily deployed and configured in a OpenStack cloud by using TripleO. OSC will be deployed in the **controller** node of the **overcloud** as shown below:  
+![](./images/osc-tripleo-services.png)  
+*OSC In the TripleO Overcloud*  
+
+## Constraints and Assumptions
+### OSC Packaged Binaries
+TripleO can only deploy OSC through a template if the OSC binaries are available in a public repository. There are multiple paths to achieve this:  
+1.  RPM package.
+2.  Docker container image on an arbitrary Docker repository.
+3.  Docker container image on a Kolla repository.
```

Comment:
How is the Kolla Image downloaded as part of heat/puppet template configuration? as i understand, if it is RPM, they use 'package' keyword' to download and run the yum install ==> as part of puppet configuration, 
How is this for Kolla image download and installation in comparison to RPM? or would this be part of separate US?

Reply:
Yes these details need to be investigated in another US, I have proposed 10 new ones :). In short TripleO infra clones/copies all the images in the Kolla repo (i.e.: https://hub.docker.com/r/kolla) to the tripleO repo (https://hub.docker.com/r/tripleoupstream/) which means that THT can reference those images directly when triggering docker run. If we do use RPM instead then yes likely we will need a separate puppet script to download and install the RPM.